models/balancer/avalanche_c/balancer_v2_avalanche_c_trades.sql

```diff
@@ -0,0 +1,206 @@
+{{ config(
+    schema = 'balancer_v2_avalanche_c',
+    tags = ['dunesql'],
+    alias = alias('trades'),
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address'],
+    post_hook = '{{ expose_spells(\'["avalanche_c"]\',
+                                "project",
+                                "balancer_v2",
+                                \'["mendesfabio", "jacektrocinski", "thetroyharris"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2023-02-17' %}
+
+WITH
+    swap_fees AS (
+        SELECT
+            swaps.poolId,
+            swaps.evt_tx_hash,
+            swaps.evt_index,
+            swaps.evt_block_number,
+            bytearray_substring(swaps.poolId, 1, 20) AS contract_address,
+            fees.swap_fee_percentage,
+            ROW_NUMBER() OVER (PARTITION BY poolId, evt_tx_hash, evt_index ORDER BY block_number DESC, index DESC) AS rn
+        FROM {{ source('balancer_v2_avalanche_c', 'Vault_evt_Swap') }} swaps
+        LEFT JOIN {{ ref('balancer_v2_avalanche_c_pools_fees') }} fees
+            ON fees.contract_address = bytearray_substring(swaps.poolId, 1, 20)
+            AND ARRAY[fees.block_number] || ARRAY[fees.index] < ARRAY[swaps.evt_block_number] || ARRAY[swaps.evt_index]
+        {% if is_incremental() %}
+        WHERE swaps.evt_block_time >= date_trunc('day', NOW() - interval '7' day)
+        {% endif %}
+    ),
+    dexs AS (
+        SELECT
+            swap.evt_block_number,
+            swap.evt_block_time AS block_time,
+            CAST(NULL AS VARBINARY) AS taker,
+            CAST(NULL AS VARBINARY) AS maker,
+            swap.amountOut AS token_bought_amount_raw,
+            swap.amountIn AS token_sold_amount_raw,
+            CAST(NULL as DOUBLE) AS amount_usd,
+            swap.tokenOut AS token_bought_address,
+            swap.tokenIn AS token_sold_address,
+            swap_fees.contract_address AS project_contract_address,
+            swap.poolId AS pool_id,
+            swap_fees.swap_fee_percentage / POWER(10, 18) AS swap_fee,
+            swap.evt_tx_hash AS tx_hash,
+            '' AS trace_address,
```

Comment:
we are deprecating this field from downstream `dex.trades`, so it can be removed from all balancer spells

if you'd prefer to take note and remove when you revisit these to update the varbinary types, that works too, then we can just ignore the field downstream. if you want to do now, that is good too

Reply:
it can be removed from unique key, unique combo test in schema files, and all throughout spells in the PR

Reply:
So trace_address can be removed from everywhere and we will be fine to merge? We won't have to wait additional time to merge?

Reply:
i will plan to merge tomorrow morning. if you have time to remove those fields before, please do. if not, please plan to do on your follow up PR to clean out data types and such

Reply:
I will do it on the follow up PR. Thank you.